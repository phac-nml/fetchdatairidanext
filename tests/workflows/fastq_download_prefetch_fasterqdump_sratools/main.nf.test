nextflow_workflow {

    name "Test workflow: fastq_download_prefetch_fasterqdump_sratools/main.nf"
    script "subworkflows/local/fastq_download_prefetch_fasterqdump_sratools/main.nf"
    workflow "FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS"
    tag "subworkflow"
    tag "subworkflow_nfcore"
    tag "custom/sratoolsncbisettings"
    tag "sratools/prefetch"
    tag "sratools/fasterqdump"
    tag "fastq_download_prefetch_fasterqdump_sratools"

    test("Parameters: default") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [[ id:'test_single_end', single_end:true  ], 'DRR000774'],
                    [[ id:'test_paired_end', single_end:false ], 'SRR11140744']
                )
                input[1] = []
                """
            }
            params {
                outdir = "output"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )

            assert path("$launchDir/output").exists()

            def lines = path("$launchDir/output/prefetch/failures_report.csv").readLines()
            assert lines.size() == 1
            assert lines.contains("sample,insdc_accession")
            assert lines.contains("test_single_end,DRR000774").equals(false)
            assert lines.contains("test_paired_end,SRR11140744").equals(false)
        }
    }

    test("Download errors: 403 and invalid") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [[ id:'SAMPLE1', single_end:false ], 'ERR1109373'],
                    [[ id:'ERROR1', single_end:false ], 'SRR999908'],
                    [[ id:'ERROR2', single_end:false ], 'INVALID!!'],
                    [[ id:'SAMPLE2', single_end:false ], 'SRR13191702']
                )
                input[1] = []
                """
            }
            params {
                outdir = "output"
            }
        }

        then {
            assert workflow.success
            assert path("$launchDir/output").exists()

            def lines = path("$launchDir/output/prefetch/failures_report.csv").readLines()
            assert lines.size() == 3
            assert lines.contains("sample,insdc_accession")
            assert lines.contains("ERROR1,SRR999908")
            assert lines.contains("ERROR2,INVALID!!")
        }
    }

}
