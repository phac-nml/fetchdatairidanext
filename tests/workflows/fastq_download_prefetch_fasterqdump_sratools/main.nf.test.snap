{
    "Parameters: default": {
        "content": [
            {
                "0": [
                    [
                        {
                            "irida_id": "test_paired_end",
                            "single_end": false
                        },
                        [
                            "SRR11140744.fastq.gz:md5,96a821b351f3a450889a38e7099f7b95",
                            "SRR11140744_1.fastq.gz:md5,8573015c91d099b6e30789f8bab2f43c",
                            "SRR11140744_2.fastq.gz:md5,37e6f719a022dc3c9994c80fbc20c311"
                        ]
                    ],
                    [
                        {
                            "irida_id": "test_single_end",
                            "single_end": true
                        },
                        "DRR000774.fastq.gz:md5,a110f93f7a9b0271455f5a435bce73c7"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1a2218ff913fc33408bffccb081b5048",
                    "versions.yml:md5,98d78bba9f3da39a0b7db6e9c7dcc224",
                    "versions.yml:md5,9c558ff624585a6eee82a19c8c0136db"
                ],
                "reads": [
                    [
                        {
                            "irida_id": "test_paired_end",
                            "single_end": false
                        },
                        [
                            "SRR11140744.fastq.gz:md5,96a821b351f3a450889a38e7099f7b95",
                            "SRR11140744_1.fastq.gz:md5,8573015c91d099b6e30789f8bab2f43c",
                            "SRR11140744_2.fastq.gz:md5,37e6f719a022dc3c9994c80fbc20c311"
                        ]
                    ],
                    [
                        {
                            "irida_id": "test_single_end",
                            "single_end": true
                        },
                        "DRR000774.fastq.gz:md5,a110f93f7a9b0271455f5a435bce73c7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1a2218ff913fc33408bffccb081b5048",
                    "versions.yml:md5,98d78bba9f3da39a0b7db6e9c7dcc224",
                    "versions.yml:md5,9c558ff624585a6eee82a19c8c0136db"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-05T15:53:20.740235848"
    }
}